tests/scripts/thread-cert/Cert_7_1_07_BorderRouterAsLeader.py

```diff
@@ -0,0 +1,567 @@
+        dut_addrs = []
+        dut_addrs.append(self.nodes[LEADER].get_addr(PREFIX_1))
+        dut_addrs.append(self.nodes[LEADER].get_addr(PREFIX_2))
+        for addr in dut_addrs:
+            self.assertTrue(self.nodes[ROUTER_1].ping(addr))
+            self.simulator.go(1)
+            self.assertTrue(self.nodes[SED].ping(addr))
+            self.simulator.go(1)
+
+        self.nodes[ROUTER_2].remove_domain_prefix(PREFIX_2)
+        self.nodes[ROUTER_2].register_netdata()
+        self.simulator.go(5)
+
+    def verify(self, pv):
+        pkts = pv.pkts
+        pv.summary.show()
+
+        LEADER = pv.vars['LEADER']
+        LEADER_RLOC = pv.vars['LEADER_RLOC']
+        LEADER_RLOC16 = pv.vars['LEADER_RLOC16']
+        ROUTER_1 = pv.vars['ROUTER_1']
+        ROUTER_1_RLOC16 = pv.vars['ROUTER_1_RLOC16']
+        ROUTER_1_RLOC = pv.vars['ROUTER_1_RLOC']
+        ROUTER_2 = pv.vars['ROUTER_2']
+        ROUTER_2_RLOC16 = pv.vars['ROUTER_2_RLOC16']
+        ROUTER_2_RLOC = pv.vars['ROUTER_2_RLOC']
+        SED = pv.vars['SED']
+        MED = pv.vars['MED']
+        GUA = [{}, {}]
+        PRE = [Bytes(PREFIX_1[:-5]), Bytes(PREFIX_2[:-5])]
+
+        for i in (0, 1):
+            for node in ('LEADER', 'ROUTER_1', 'SED'):
+                for addr in pv.vars['%s_IPADDRS' % node]:
+                    if addr.startswith(PRE[i]):
+                        GUA[i][node] = addr
+
+        # Step 1: Ensure topology is formed correctly
+        pv.verify_attached('ROUTER_1', 'LEADER')
+        pv.verify_attached('ROUTER_2', 'LEADER')
+        pv.verify_attached('MED', 'LEADER', 'MTD')
+        pv.verify_attached('SED', 'LEADER', 'MTD')
+        _pkt = pkts.last()
+
+        # Step 2,3: Router_1 and Router_2 MUST send a CoAP Server Data
+        #           Notification frame to the Leader including the server’s
+        #           information(Prefix, Border Router):
+        #             CoAP Request URI
+        #                 coap://[<Leader address>]:MM/a/sd
+        #             CoAP Payload
+        #                 Thread Network Data TLV
+
+        # Step 4: Leader sends a CoAP ACK frame to each of Router_1 and
+        #         Router_2
+        with pkts.save_index():
+            for node in ('ROUTER_1', 'ROUTER_2'):
+                pkts.filter_wpan_src64(pv.vars['%s' %node]).\
+                    filter_wpan_dst16(LEADER_RLOC16).\
+                    filter_coap_request(SVR_DATA_URI).\
+                    filter(lambda p:
+                           [Ipv6Addr(PREFIX_1[:-3])] ==
+                           p.thread_nwd.tlv.prefix and\
+                           [pv.vars['%s_RLOC16' %node]] ==
+                           p.thread_nwd.tlv.border_router_16
+                           ).\
+                    must_next()
+                pkts.filter_wpan_src64(LEADER).\
+                    filter_ipv6_dst(pv.vars['%s_RLOC' %node]).\
+                    filter_coap_ack(SVR_DATA_URI).\
```

Comment:
better also check that `Message ID` is equal to the CoAP request.

Reply:
added the check